x-pack/plugins/licensing/server/license.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { createHash } from 'crypto';
+import { LicenseFeature } from './license_feature';
+import { LICENSE_STATUS, LICENSE_TYPE } from './constants';
+import { LicenseType, ILicense } from './types';
+
+function toLicenseType(minimumLicenseRequired: LICENSE_TYPE | string) {
+  if (typeof minimumLicenseRequired !== 'string') {
+    return minimumLicenseRequired;
+  }
+
+  if (!(minimumLicenseRequired in LICENSE_TYPE)) {
+    throw new Error(`${minimumLicenseRequired} is not a valid license type`);
+  }
+
+  return LICENSE_TYPE[minimumLicenseRequired as LicenseType];
+}
+
+export class License implements ILicense {
+  private readonly hasLicense: boolean;
+  private readonly license: any;
+  private readonly features: any;
+  private _signature!: string;
+  private objectified!: any;
+  private readonly featuresMap: Map<string, LicenseFeature>;
+
+  constructor(
+    license: any,
+    features: any,
+    private error: Error | null,
+    private clusterSource: string
+  ) {
+    this.hasLicense = Boolean(license);
+    this.license = license || {};
+    this.features = features;
+    this.featuresMap = new Map<string, LicenseFeature>();
+  }
+
+  public get uid() {
+    return this.license.uid;
+  }
+
+  public get status() {
+    return this.license.status;
+  }
+
+  public get isActive() {
+    return this.status === 'active';
+  }
+
+  public get expiryDateInMillis() {
+    return this.license.expiry_date_in_millis;
+  }
+
+  public get type() {
+    return this.license.type;
+  }
+
+  public get isAvailable() {
+    return this.hasLicense;
+  }
+
+  public get isBasic() {
+    return this.isActive && this.type === 'basic';
+  }
+
+  public get isNotBasic() {
+    return this.isActive && this.type !== 'basic';
+  }
+
+  public get reasonUnavailable() {
+    if (!this.isAvailable) {
+      return `[${this.clusterSource}] Elasticsearch cluster did not respond with license information.`;
+    }
+
+    if (this.error instanceof Error && (this.error as any).status === 400) {
```

Comment:
who sets this `status` field? cannot see it on Boom errors

Reply:
This was a carry-over from the previous implementation, so I'm not sure who sets this. Any recommendations on an alternative?